release_notes/ocp-4-12-release-notes.adoc

```diff
@@ -341,6 +341,49 @@ The Cloud Credential Operator utility (`ccoctl`) now creates secrets that use re
 
 With this release, when you xref:../installing/installing_gcp/uninstalling-cluster-gcp.adoc#cco-ccoctl-deleting-sts-resources_uninstalling-cluster-gcp[delete GCP resources with the Cloud Credential Operator utility], you must specify the directory containing the files for the component `CredentialsRequest` objects.
 
+[discrete]
+[id="ocp-4-12-psa-restricted-enforcement"]
+=== Future restricted enforcement for pod security admission
+
+Currently, pod security violations are shown as warnings and logged in the audit logs, but do not cause the pod to be rejected.
+
+Global restricted enforcement for pod security admission is currently planned for the next minor release of {product-title}. When this restricted enforcement is enabled, pods with pod security violations will be rejected.
+
+See xref:../authentication/understanding-and-managing-pod-security-admission.adoc#security-context-constraints-psa-alert-eval_understanding-and-managing-pod-security-admission[Identifying pod security violations] for information about how to find which workloads are causing pod security violations.
+
+// TODO: What "call to action" can we say after this for people to prep for 4.13 enforcement? I don't see any docs on what to actually do when you have a workloads in violation
+
+// Resources
+// - https://connect.redhat.com/blog/important-openshift-changes-pod-security-standards
+// - https://docs.google.com/document/d/1fL8kDkAGejFM335AwjFL2A0keIKbQEQ2kZYNVL7FTiU/edit#
+
+To prepare for this upcoming change, ensure that your workloads can run as restricted. Workloads that are not configured according to the enforced security standards defined globally or at the namespace level will be rejected. The `restricted-v2` SCC admits workloads according to the link:https://kubernetes.io/docs/concepts/security/pod-security-standards/#restricted[Restricted] Kubernetes definition.
+
+However, if your workload requires escalated permissions, ... TODO
```

Comment:
@stlaz I read through a bunch of resources to try to find out what the potential actions are that users might need to take.  But it seemed like a lot of options and that the actions could vary depending on the situation.

Anyways, I'm looking for what we could say here, that if they do need escalated permissions, what do they do? I have some notes in the comments below.

Do they do nothing as long as the label syncer is enabled for that namespace? Yes, only as long as xyz other condition is met?

When do they add a label to the namespace? Or when would they want to grant access to the restricted SCC?

We can chat about this over slack tomorrow if that's easier.

Also note that this PR is just for the release notes, so we'd want to keep it pretty concise. If we need to get into more details, we could consider adding a section to the main docs on PSA. Or link to kbase articles if we have them. This one on granting access to the restricted SCC looks like it could be helpful if we go the linking route: https://access.redhat.com/articles/6973044

Reply:
The hard thing about this is that the actions one need to take are very individual.

I believe that if they notice that these violations are caused by vendored software, they should check with the vendor and their documentation.

If the issues are caused by software that they developed themselves, they need to
1. read about OpenShift autolabelling and check why it's not enough for them (running in openshift-* namespaces, creating pods directly without a pod controller, ...)
2. carefully read in the violation message why their pods are violating the pod security profile
3. read https://kubernetes.io/docs/concepts/security/pod-security-admission/ and https://docs.openshift.com/container-platform/4.11/authentication/understanding-and-managing-pod-security-admission.html#security-context-constraints-psa-opting_understanding-and-managing-pod-security-admission to understand how they can set custom admission profile if necessary
4. change the pod or the namespace admission profile